### ISSUE TITLE
entropy_from_logits returns NaN for zero-probability tokens

### ISSUE THREAD

**Issue by vineethsaivs** [huggingface/trl#7221]
## Reproduction

`entropy_from_logits` returns NaN when a vocabulary entry has zero probability. This also happens with finite float16 logits when log-softmax reaches negative infinity. The categorical distribution is valid: the zero-probability term should contribute zero to Shannon entropy.

```python
import torch
from trl.trainer.utils import entropy_from_logits

cases = [
    torch.tensor([[0.0, float("-inf")]]),
    torch.tensor([[0.0, 0.0, float("-inf")]]),
    torch.tensor([[65504.0, -65504.0]], dtype=torch.float16),
]
for logits in cases:
    actual = entropy_from_logits(logits)
    reference = torch.distributions.Categorical(logits=logits).entropy()
    print(logits.dtype, "TRL:", actual.tolist(), "Categorical:", reference.tolist())
```

Output:

```text
torch.float32 TRL: [nan] Categorical: [-0.0]
torch.float32 TRL: [nan] Categorical: [0.6931471824645996]
torch.float16 TRL: [nan] Categorical: [-0.0]
```

The multiplication `exp(logps) * logps` evaluates `0 * -inf`. PyTorch's categorical entropy returns finite results for the same inputs. This helper is used for entropy metrics in the trainers.

## System Info

```text
macOS-26.5.2-arm64-arm-64bit
Python 3.12.10
PyTorch 2.10.0
Transformers 5.17.0
TRL 1.14.0.dev0
Device: CPU
```

Reproduced from main at `03ad26d2c56974b2e72e57fcb7e5002b47b55ca5`. No model downloads or accelerator are needed.

## Checklist

- [x] I checked existing issues and PRs for this failure
- [x] System information is included
- [x] The reproduction is minimal and complete
- [x] Code and output use code blocks
- [x] No exception is raised; the incorrect result is shown above

AI assistance was used to investigate and reproduce this report.

**Comment by qgallouedec** [comment]
Hey @vineethsaivs, the arithmetic is right (`0 * -inf` is NaN, and PyTorch clamps to `finfo.min` where we don't), but I can't find a path that puts a `-inf` logit in there. Every caller hands the helper `outputs.logits` straight from a forward pass.

Please provide a real training config that hit this.

**Comment by jnuthong** [comment]
I'd like to work on this. 
Will open a PR shortly.

**Comment by jnuthong** [comment]
Opened PR #7251 to fix this. Root cause: `0 * -inf` in the Shannon entropy sum for zero-probability tokens. Fix treats those terms as 0, matching `torch.distributions.Categorical.entropy()`, with regression tests from the reproducers in this issue.

**Comment by jnuthong** [comment]
Reopened with a new PR following the required template: #7252 (previous PR #7251 was closed for missing template sections).

**Comment by qgallouedec** [comment]
Closing as not evidence real training reaches it
